include/deal.II/grid/grid_tools.h

```diff
@@ -1032,6 +1032,35 @@ namespace GridTools
   ( const MeshType                                                                    &mesh,
     const std::function<bool (const typename MeshType::active_cell_iterator &)> &predicate );
 
+  /**
+   * Compute a collection of bounding boxes so that all cells for which the given predicate is true, are
+   * completely enclosed in at least one of the bounding boxes.
+   *
+   * The algorithm loops over all cells at given level @p refinement_level and active cells at coarser refinement
+   * levels, creating a Bounding Box for each of them. Then it merges the boxes which can be expressed as a single
+   * one. In a last optional step all smaller boxes are merged with neighbors.
+   *
+   * The parameters to control the algorithm are:
+   * - @p predicate : the property of the cells to enclose e.g. IteratorFilters::LocallyOwnedCell .
+   *  The predicate is tested only on active cells.
+   * - @p refinement_level : it defines the level at which the initial bounding box are created. The refinement
+   *  should be set to a coarse refinement level. If @p refinement_level is higher than the number of levels of the
+   *  triangulation an exception is thrown.
+   * - @p max_boxes : the maximum number of bounding boxes to compute. If more are created the smaller ones are
+   *  merged with neighbors. By default after merging the boxes which can be expressed as a single one no
+   *  more boxes are merged. See the BoundingBox::is_neighbor() function for details.
+   *  Notice only neighboring cells are merged (see the @p is_neighbor function in bounding box class): if
+   *  the target number of bounding boxes max_boxes can't be reached by merging neighbors an exception is thrown
```

Comment:
are you trying to say that the set of boxes is not an exact cover but may contain additional space?

Reply:
By default, unless one uses the most refined level to create the bounding boxes (in which case these boxes shall occupy the same space as cells), the cover probably contains additional space.
The function isn't meant to describe precisely the area/volume occupied by cells with predicate, but to find a good enough approximation.